### PR TITLE
Improvements to Admin/Provider

### LINKED ISSUE #3463: HTTPS checkbox label is wrong
Checkbox for HTTPS is wrongly written as "Use HTTP", the opposite of what it actually does.
Screenshot 2022-05-30 at 15 36 58

### LINKED ISSUE #3482: Remove delete-button from providers page

### LINKED ISSUE #3483: Make domain/port of provider editable

### PR DESCRIPTION
**Pull request by xiroV**
Fixes to admin/provider page, and functionality to edit/change provider domain and port.

fixes #3482 
fixes #3483 
fixes #3463

### CODE REVIEW

File: backend/accounting-service/src/jvmMain/kotlin/services/providers/ProviderService.kt [hunk 1]
```diff
@@ -281,4 +281,33 @@ class ProviderService(
             throw ex
         }
     }
+
+    suspend fun update(
+        actorAndProject: ActorAndProject,
+        request: ProvidersUpdateSpecificationRequest,
+        ctx: DBContext = db
+    ): ProvidersUpdateSpecificationResponse {
+        val (actor) = actorAndProject
```
Comment (DanThrane) [inline backend/accounting-service/src/jvmMain/kotlin/services/providers/ProviderService.kt:290]: This should be editable by anyone who has "EDIT" rights to the provider (the resource). You can use the `retrieveBulk` call for this.

File: frontend-web/webclient/app/Admin/Providers/Edit.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                        }
+                    })
+                }}
+                onSubmitSucceded={(res, data) => {
+                    if (res) {
+                        history.push(`/providers`);
+                    }
+                }}
+            >
+                <RS.Text
```
Comment (DanThrane) [inline frontend-web/webclient/app/Admin/Providers/Edit.tsx:61]: I think we should aim for re-using this component across edit and create.

File: frontend-web/webclient/app/UCloud/ProvidersApi.tsx [hunk 3]
```diff
@@ -92,13 +108,67 @@ class ProviderApi extends ResourceApi<Provider, Product, ProviderSpecification,
...
 
+    public retrieveOperations(): Operation<Provider, ResourceBrowseCallbacks<Provider> & ProviderCallbacks>[] {
+        return [
+            {
+                text: "Create " + this.title.toLowerCase(),
+                icon: "upload",
+                color: "blue",
+                primary: true,
+                canAppearInLocation: loc => loc !== "IN_ROW",
+                enabled: (selected, cb) => {
```
Comment (DanThrane) [inline frontend-web/webclient/app/UCloud/ProvidersApi.tsx:119]: Should only be enabled if you are a UCloud admin (you can use this UI if you have edit rights for the provider)

File: frontend-web/webclient/app/UCloud/ProvidersApi.tsx [hunk 4]
```diff
@@ -92,13 +108,67 @@ class ProviderApi extends ResourceApi<Provider, Product, ProviderSpecification,
...
+                    if (selected.length !== 0 || cb.startCreation == null || cb.isCreating) return false;
+                    return true;
+                },
+                onClick: (selected, cb) => cb.startCreation!(),
+                tag: CREATE_TAG
+            },
+            {
+                text: "Edit",
```
Comment (DanThrane) [inline frontend-web/webclient/app/UCloud/ProvidersApi.tsx:127]: Should be enabled if you have edit rights for the resource.

**Review by DanThrane — CHANGES_REQUESTED** [review]
Looks mostly good to me but I have a few changes.

File: frontend-web/webclient/app/Admin/Providers/Edit.tsx (hunk 2 above)
Comment (Tiggles) [inline frontend-web/webclient/app/Admin/Providers/Edit.tsx:61]: Any reason not to omit this? You can just reuse `provider.data?.specification.id` in `createRequest`, can't you?

File: frontend-web/webclient/app/UCloud/ProvidersApi.tsx [hunk 5]
```diff
@@ -204,6 +274,19 @@ class ProviderApi extends ResourceApi<Provider, Product, ProviderSpecification,
...
+    ): APICallParameters<ProvidersUpdateSpecificationRequest, any /* unknown */> {
+        return {
+            context: "",
+            method: "POST",
+            path: "/api/providers" + "/update",
+            parameters: request,
+            reloadId: Math.random(),
+            payload: request,
+        };
+    }
```
Comment (Tiggles) [inline frontend-web/webclient/app/UCloud/ProvidersApi.tsx:288]: @DanThrane, I haven't used it much, but `apiUpdate` should replace the object created here, right?

**Review by DanThrane — APPROVED** [review]

**Pull request merged**